lang/test/org/partiql/lang/syntax/SqlParserTest.kt

```diff
@@ -3951,6 +3950,44 @@ class SqlParserTest : SqlParserTestBase() {
         )
     }
 
+    //****************************************
+    // OFFSET clause parsing
+    //****************************************
+
+    private fun buildProject(project: String) = PartiqlAst.build { projectList(projectExpr(id(project))) }
+
+    private fun buildLit(lit: String) = PartiqlAst.Expr.Lit(loadSingleElement(lit))
+
+    @Test
+    fun selectOffsetTest() = assertExpression("SELECT x FROM a OFFSET 5") {
+        select(
+            project = buildProject("x"),
+            from = scan(id("a")),
+            offset = buildLit("5")
+        )
+    }
+
+    @Test
+    fun selectLimitOffsetTest() = assertExpression("SELECT x FROM a LIMIT 7 OFFSET 5") {
+        select(
+            project = buildProject("x"),
+            from = scan(id("a")),
+            limit = buildLit("7"),
+            offset = buildLit("5")
+        )
+    }
+
+    @Test
+    fun selectWhereLimitOffsetTest() = assertExpression("SELECT x FROM a WHERE y = 10 LIMIT 7 OFFSET 5") {
+        select(
+            project = buildProject("x"),
+            from = scan(id("a")),
+            where = PartiqlAst.Expr.Eq(listOf(id("y"), buildLit("10"))),
+            limit = buildLit("7"),
+            offset = buildLit("5")
+        )
+    }
+
```

Comment:
Would be good to add a test using `ORDER BY`, which will be the most common use case when using `OFFSET`